yarn-project/types/package.json

```diff
@@ -31,6 +31,7 @@
   },
   "dependencies": {
     "@aztec/circuits.js": "workspace:^",
+    "@aztec/ethereum": "workspace:^",
```

Comment:
`aztec/ethereum` brings a bunch of annoying L1 dependencies (such as our l1-artifacts) that we shouldn't need to pull in everywhere, given `types` is imported everywhere. It'd be great if we can remove this new dependency. 

I think we should be able to roll back L1ContractAddresses to be just a type or interface (and define it within `types`), and let the converter work its magic by working on a plain object:

https://github.com/AztecProtocol/aztec-packages/blob/471159bfbbcb13a1f26f3613ed431d929efea59b/yarn-project/foundation/src/json-rpc/convert.test.ts#L62-L66